### ISSUE TITLE
Out of memory and SIGSEGV on arch linux

### ISSUE THREAD

**Issue by jasonrbriggs** [nim-lang/nimble#352]
I'm suddenly getting the following errors running nimble from inside my project dir:

```
$ nimble refresh
out of memory
$ nimble install
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
$ nimble build
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
$ nimble install docopt
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

It was fine up until yesterday, and I'm not sure exactly what changed -- the only thing I remember changing in the project nimble file was adding untar as a dependency, but removing that makes no difference. I've also tried removing the ~/.nimble directory as well - didn't help.  
Outside of the project dir, it seems to work okay (i.e. docopt installed fine from home dir) - `nim c test.nim` also works fine.

The error looks similar to this: https://github.com/nim-lang/nimble/issues/68, but that was back in 2014, so assuming not related.

Here's my test.nimble file -- nothing too exciting:

```
# Package
version       = "0.2.0"
author        = "Jason R Briggs"
description   = "Test"
license       = "todo"

bin           = @["test"]

# Dependencies

requires @["nim >= 0.16.0", "docopt", "midnight_dynamite", "python3", "untar" ]
```

Any clues as to what I might've done wrong?

**Comment by jasonrbriggs** [comment]
PS. And if I leave it alone for a while, come back it works again. Which is even weirder...

**Comment by jasonrbriggs** [comment]
Think I've figured out the problem. I've been putting some test files into <projectdir>/tmp -- yesterday I copied in about 30K worth of rubbish files to try something out. Looks like nimble is reading that in and then blowing chunks as a consequence.  Interestingly, it doesn't matter where I put them in the project - it still causes the problem.  Is there a way to tell nimble - ignore this directory?

**Comment by dom96** [comment]
Interesting. What test files did you put into /tmp? It would be interesting to get a stack trace from Nimble but you'd need to compile a debug version of Nimble yourself for that (if you're willing to do that it would help a lot). Otherwise you can give me some more info to help me reproduce the issue.

**Comment by jasonrbriggs** [comment]
Basically I dumped the contents of a mathjax archive (https://github.com/mathjax/MathJax/archive/master.zip) into the tmp dir of the project, and after that I hit the out of mem/sigsegv issue.  Roughly 30,000 files.

**Comment by dom96** [comment]
You sure that your /tmp wasn't just filled up? Why do you think Nimble was reading these files?

**Comment by dom96** [comment]
IIRC Arch Linux uses your RAM for /tmp so filling it up could have caused out of memory errors.

**Comment by jasonrbriggs** [comment]
Not /tmp.  I meant: /home/jasonbriggs/dev/myproject/tmp

And it doesn't matter whether the files are in "myproject/tmp" or just dumped in any subdir of "myproject".  Same issue still occurs.

**Comment by jangko** [comment]
I encountered similar issue, not exactly the same, but when `nimble refresh` was called, the result was
```
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```
and no OOM,

also this happened only in appveyor env. This is my appveyor [script](https://gist.github.com/jangko/cd4ffa5c9de1e836d219832796880306)

an this is the appveyor [log](https://ci.appveyor.com/project/jangko/nimagg/build/96).

don't know what happened here, I cannot reproduce it on my computer.

**Comment by dom96** [comment]
@jangko can you reproduce it on App veyor or is it random? If you could get a stack trace out of Nimble that would help out a lot.

**Comment by jangko** [comment]
when I changed my appveyor scipt to (this)[https://github.com/jangko/nimAGG/blob/enhancement/appveyor.yml], the problem is gone. so I cannot reproduce it and cannot get the stack trace too.

what have I done is just copying the koch nimble's related code and remove -d:release switch with the hope I can get the stack trace, but the problem is gone.

previously I use koch nimble.

**Comment by dom96** [comment]
I'm hoping that this is fixed now thanks to #385. Reopen if not.
